### ISSUE TITLE
Add benchmark CI: Debian

### ISSUE THREAD

**Issue by kou** [apache/arrow-go#85]
This part: https://github.com/apache/arrow/blob/8113594c85b4c578236cdd69d4155e0e118744b4/.github/workflows/go.yml#L136-L163

**Comment by singh1203** [comment]
Hello @kou, I would love to work on this issue. Looking at the description, I understand it involves implementing a benchmarking CI workflow for the arrow-go repository using Debian as the runtime environment.
Correct me if I am wrong.
Thank you.

**Comment by kou** [comment]
See also: apache/arrow44996

**Comment by assignUser** [comment]
I think we should keep running the go builds on github runners, the setup and maintenance is just much less complicated. As in the monorepo we can still upload the results to conbench. When the PR for this is ready I can handle creating the secrets and open an infra ticket to get them set in the repo.

**Comment by raulcd** [comment]
> I think we should keep running the go builds on github runners

Do you mean start using GitHub runners instead of the dedicated buildkite infra to run this new benchmarks?
Even though this will probably make the benchmark reports slightly more unstable I do agree that given the current status it would be better. If we ever want to have dedicated runners we can discuss this with Infra maybe?

**Comment by assignUser** [comment]
Yes. Go has never used the buildkite infra and I think it should work for the rest as well and make everything more transparent to the community.

**Comment by singh1203** [comment]
> Yes. Go has never used the buildkite infra and I think it should work for the rest as well and make everything more transparent to the community.

Okay, I think it's a green signal for me to start working on it and to be more specific I should add the Benchmark job in the test runner file to leverage :

- Reuse of Configuration
- Centralized Management
- Fewer files to manage

However, in my opinion, it increases complexity because benchmarks differ from test purposes.
